### PR TITLE
Adding VC project properties info to 'how to resolve' suggestions.

### PR DESCRIPTION
**Pull request by olgaark**
Adding VC project properties info to suggestions
Making long strings multiline for easier reading.
Adding more info for BA2004 and clarifying some suggestions for compiler defaults.

### CODE REVIEW

File: src/BinSkim.Rules/RuleResources.resx [hunk 1]
```diff
@@ -266,43 +310,63 @@ Modules triggering this check were:
...
+To resolve this issue, don't set /HIGHENTROPYVA:NO on link.exe command line and allow it to be enabled by default. (This image was determined to have been properly compiled as /LARGEADDRESSAWARE.)</value>
   </data>
   <data name="BA2015_Error_NoLargeAddressAware" xml:space="preserve">
-    <value>'{0}' does not declare itself as high entropy ASLR compatible. High entropy makes Address Space Layout Randomization more effective in mitigating memory corruption vulnerabilities. To resolve this issue, configure your tools to mark the program high entropy compatible by supplying /LARGEADDRESSAWARE to the C or C++ linker command line. (This image was determined to have been properly compiled as /HIGHENTROPYVA.)</value>
+    <value>'{0}' does not declare itself as high entropy ASLR compatible. High entropy makes Address Space Layout Randomization more effective in mitigating memory corruption vulnerabilities.
+To resolve this issue, don't set /LARGEADDRESSAWARE:NO on link.exe command line. (This image was determined to have been properly compiled as /HIGHENTROPYVA.)</value>
```
Comment (shaopeng-gh) [inline src/BinSkim.Rules/RuleResources.resx:341]: >. [](http://example.com/codeflow?start=79&length=1)

should we use same style as above with `and allow it to be enabled by default.`

File: src/BinSkim.Rules/RuleResources.resx [hunk 2]
```diff
@@ -266,43 +310,63 @@ Modules triggering this check were:
...
   </data>
   <data name="BA2018_EnableSafeSEH_Description" xml:space="preserve">
-    <value>X86 binaries should enable the SafeSEH mitigation to minimize exploitable memory corruption issues. SafeSEH makes it more difficult to exploit vulnerabilities that permit overwriting SEH control blocks on the stack, by verifying that the location to which a thrown SEH exception would jump is indeed defined as an exception handler in the source program (and not shellcode). To resolve this issue, supply the /SafeSEH flag on the linker command line. Note that you will need to configure your build system to supply this flag for x86 builds only, as the /SafeSEH flag is invalid when linking for ARM and x64.</value>
+    <value>X86 binaries should enable the SafeSEH mitigation to minimize exploitable memory corruption issues. SafeSEH makes it more difficult to exploit vulnerabilities that permit overwriting SEH control blocks on the stack, by verifying that the location to which a thrown SEH exception would jump is indeed defined as an exception handler in the source program (and not shellcode).
+To resolve this issue, supply the /SafeSEH flag on link.exe command line.
```
Comment (shaopeng-gh) [inline src/BinSkim.Rules/RuleResources.resx:361]: >link.exe command line [](http://example.com/codeflow?start=51&length=21)

should we change all instance of `link.exe` in the PR to just `linker`.
For user that is not using Microsoft toolchain.

File: src/BinSkim.Rules/RuleResources.resx [hunk 3]
```diff
@@ -266,43 +310,63 @@ Modules triggering this check were:
...
+    <value>'{0}' is an x86 binary which {1}, indicating that it does not enable the SafeSEH mitigation. SafeSEH makes it more difficult to exploit memory corruption vulnerabilities that can overwrite SEH control blocks on the stack, by verifying that the location to which a thrown SEH exception would jump is indeed defined as an exception handler in the source program (and not shellcode).
+To resolve this issue, supply the /SafeSEH flag on link.exe command line.
+For VC projects use ItemDefinitionGroup - Link = ImageHasSafeExceptionHandlers property with 'true' value.
```
Comment (shaopeng-gh) [inline src/BinSkim.Rules/RuleResources.resx:368]: >VC projects [](http://example.com/codeflow?start=4&length=11)

For user that is not using Microsoft toolchain. Do we think 
`VC projects` is clear for them or we want a longer name more clear,
I am not sure, an open question.

**Comment by shaopeng-gh** [comment]
Thanks for submitting this PR!

File: src/BinSkim.Rules/RuleResources.resx (hunk 2 above)
Comment (olgaark) [reply src/BinSkim.Rules/RuleResources.resx:361]: Most of the switches are tool specific, clang/gcc have their own, not /SafeSEH.

Comment (olgaark) [reply src/BinSkim.Rules/RuleResources.resx:361]: /SafeSEH is only applicable to link.exe, clang/GCC have different switches.

File: src/BinSkim.Rules/RuleResources.resx (hunk 3 above)
Comment (olgaark) [reply src/BinSkim.Rules/RuleResources.resx:368]: What would you suggest? "VC" is short name for "Visual Studio C++" and if you search for "VC Project" you get
https://learn.microsoft.com/en-us/cpp/build/creating-and-managing-visual-cpp-projects?view=msvc-170. 
Alternatively, I can say "for vcxproj" - would it be better?

File: src/BinSkim.Rules/RuleResources.resx (hunk 2 above)
Comment (shaopeng-gh) [reply src/BinSkim.Rules/RuleResources.resx:361]: Yes. just a note not related to this PR, imo we have assumptions in a lot places the user is using Microsoft toolchain, while BinSkim target is just PE files.

File: src/BinSkim.Rules/RuleResources.resx (hunk 3 above)
Comment (shaopeng-gh) [reply src/BinSkim.Rules/RuleResources.resx:368]: Just tested "VC Project" with non-Microsoft search engine with Incognito, the fist 3 results all are something else other than the topic. The Microsoft search engine did return all related topic. 
But "VC Project" is clear as well.
+ @suvamM if any preference.

**Review by shaopeng-gh — APPROVED** [review]
:shipit:

File: src/BinSkim.Rules/RuleResources.resx (hunk 3 above)
Comment (olgaark) [reply src/BinSkim.Rules/RuleResources.resx:368]: How about I leave it as "VC project" for now? As soon as people can quickly tell if this info is applicable to them or not, we are good here. And we can always search/replace it with something else if we get feedback.

Comment (shaopeng-gh) [reply src/BinSkim.Rules/RuleResources.resx:368]: sounds good to me

**Comment by michaelcfanning** [comment]
Our pipeline won't run against a code fork for security reasons. I've added you as a contributor, would you mind recreating your PR as a direct branch off main? :)

**Pull request closed without merging**